drivers/flash/spi_nor.c

```diff
@@ -760,11 +767,19 @@ static int spi_nor_read(const struct device *dev, off_t addr, void *dest,
 		return -EINVAL;
 	}
 
+	/* Ensure flash is powered before read */
+	if (pm_device_runtime_get(dev) < 0) {
+		return -EIO;
+	}
+
 	acquire_device(dev);
 
 	ret = spi_nor_cmd_addr_read(dev, SPI_NOR_CMD_READ, addr, dest, size);
 
 	release_device(dev);
+
+	/* Release flash power requirement */
+	(void)pm_device_runtime_put_async(dev, K_MSEC(DPD_INACTIVITY_TIMEOUT_MS));
```

Comment:
If DPD_INACTIVITY_TIMEOUT_MS is 0, should this call pm_device_runtime_put() instead of put_async()? Otherwise it puts the work on the system workqueue without delay, which seems like needless extra work.

Reply:
This sort of optimization is better suited to occur in the PM implementation so that all users get the benefits and code isn't copy pasted everywhere.